src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,48 @@
-import React from 'react';
+import React, {
+  Dispatch,
+  SetStateAction,
+  useMemo,
+  useState,
+} from 'react';
+import cn from 'classnames';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  todos: Todo[],
+  isModalLoading: boolean;
+  selectedTodoId: number;
+  setIsModalLoading: Dispatch<SetStateAction<boolean>>;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  todos,
+  selectedTodoId,
+  isModalLoading,
+  setIsModalLoading,
+  onClose,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  const selectedTodo: Todo = todos.find(
+    todo => todo.id === selectedTodoId,
+  ) || todos[0];
```

Comment:
You can store `selectedTodo` itself in the App component instead of `selectedTodoId` so you don't need to find it one more time